test/aws/client_test.exs

```diff
@@ -74,7 +74,69 @@ defmodule AWS.ClientTest do
       # Bypass should not assert on the request having been completed.
       Bypass.pass(bypass)
     end
+
+    test "retries failed requests", %{client: client, bypass: bypass} do
+      # Start the Agent
+      TestRetryCounter.start_link(0)
```

Comment:
Oh, nice strategy to use a counter Agent here :)
I think it would be better to work with the `pid` instead of having a named process. This is because with the `async: true` of the test, it could increment while the other test is running, thus affecting the other test.

Just to complement, there is the `:counters` API from Erlang that could be useful here as well: https://www.erlang.org/doc/man/counters
The advantage of `:counters` is that you don't need to create a process for that.

Here you would:

```elixir
counter = :counters.new(1, [:atomics])
```

And to increment you would:

```elixir
:counters.add(counter, 1, 1)
```

To get the results, it's just `:counters.get(counter, 1)`.
WDYT?

Reply:
This is great! Thanks! Changed!